src/NetMQ.Tests/NetMQPollerTest.cs

```diff
@@ -400,6 +402,150 @@ public void RemoveSocket()
             }
         }
 
+        [Fact]
+        public async void RemoveAndDisposeSocket()
```

Comment:
These two new tests could be refined a little, and at least a helper method to do the boiler plate setup in on place...